tests/lib/external/prepare-ssh.sh

```diff
@@ -3,17 +3,39 @@ set -ex
 
 INSTANCE_IP="${1:-localhost}"
 INSTANCE_PORT="${2:-8022}"
-USER="${3:-$(whoami)}"
+USER_NAME="${3:-$(whoami)}"
+USER_PASS="${4:-}"
+USER_TYPE="${5:-}"
+
+SSHPASS_PARAM=""
 
 execute_remote(){
     # shellcheck disable=SC2029
-    ssh -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -p "$INSTANCE_PORT" "$USER@$INSTANCE_IP" "$@"
+    $SSHPASS_PARAM ssh -q -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no -p "$INSTANCE_PORT" "$USER_NAME@$INSTANCE_IP" "$@"
 }
 
-execute_remote "sudo adduser --uid 12345 --extrausers --quiet --disabled-password --gecos '' test"
-execute_remote "echo test:ubuntu | sudo chpasswd"
-execute_remote "echo 'test ALL=(ALL) NOPASSWD:ALL' | sudo tee /etc/sudoers.d/create-user-test"
+USER_TYPE_PARAM=""
+if [ -z "$USER_TYPE" ]; then
+    USER_TYPE_PARAM="--extrausers"
+fi
+
+# It is needed to use sshpass to connect using password
+if [ -n "$USER_PASS" ]; then
+    SSHPASS_PARAM="sshpass -p $USER_PASS"
+fi
+
+# When user is root, neither sudo nor sshpass is needed
+SUDO_PARAM="sudo"
+if [ "$USER_NAME" = "root" ]; then
+    SUDO_PARAM=""
+elif [ -n "$USER_PASS" ]; then
+	SUDO_PARAM="sshpass -p $USER_PASS sudo"
```

Comment:
could you also add a comment here:

```suggestion
	# we need to use sudo to switch from the user to root, but the user has a password
	# so we also need to use sshpass to provide the password to sudo
	SUDO_PARAM="sshpass -p $USER_PASS sudo"
```